components/remote_settings/src/remote_settings.udl

```diff
@@ -73,3 +77,50 @@ interface RemoteSettings {
     [Throws=RemoteSettingsError]
     void download_attachment_to_path(string attachment_id, string path);
 };
+
+/// Stores application-level configuration data and constructs clients
+interface RemoteSettingsService {
+    constructor(RemoteSettingsConfig2 config);
+
+    /// Construct a new client with the configuration passed to the constructor
+    [Throws=RemoteSettingsError]
+    RemoteSettingsClient make_client(string collection_name);
+};
+
+/// Configuration for RemoteSettingsService
+///
+/// Once all clients have moved to the new API, we can remove the old RemoteSettingsConfig and
+/// rename this.
+dictionary RemoteSettingsConfig2 {
+    /// Server to use
+    RemoteSettingsServer server;
+    /// Directory to store data in.  Must be writable by the client.
+    string storage_dir;
+    /// Override bucket name ("main" by default, "preview" for preview collections)
+    string? bucket_name = null;
+};
+
+/// New remote settings client
+///
+/// This is a WIP matches the semantics of the official client more closely.  Once this is complete,
+/// we may decide to deprecate and remote the RemoteSettings class.
+interface RemoteSettingsClient {
+    /// Get the current set of records.
+    ///
+    /// If no records have been downloaded, this will return null.  Use `sync` to fetch the current
+    /// records from the server.
+    [Throws=RemoteSettingsError]
+    sequence<RemoteSettingsRecord>? get();
+
+    /// Sync records with the server.
+    [Throws=RemoteSettingsError]
+    void sync();
+
+    /// Sync records with the server if there are no records currently persisted.
+    [Throws=RemoteSettingsError]
+    void sync_if_empty();
+
+    /// Download an attachment with the provided id to the provided path.
+    [Throws=RemoteSettingsError]
+    void download_attachment_to_path(string attachment_id, string path);
```

Comment:
Desktop currently has a slightly different interface. It downloads the attachment to the database (if not included in dumps locally), and then either allows loading from the dump, the database cache or to actually download it. 

It also passes in the attachment record, which I believe means it can check hashes and out-of-date.

What's the advantage of downloading this to application specified file? Would it be better to have remote settings handle the file paths/cache, and then maybe tell the app where the path of the file is?

The desktop api currently returns blobs, but maybe storing the file under storage_path and returning the path within that would be better?

Reply:
On Desktop we deprecated the file-base API, because it was tedious to purge old files etc. 
If there are uniffi concerns, maybe we could have a hybrid approach, where downloaded files are tracked in a db and referenced as paths?

Reply:
Let's just remove this method for now, I don't know of a real use-case for it.